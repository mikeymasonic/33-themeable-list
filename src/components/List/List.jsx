import React from 'react';
import PropTypes from 'prop-types';
// import { Link } from 'react-router-dom';

const List = ({ id, name, species, status, image }) => (
  <section>
    {/* <h4><Link to={`detail/${id}`} >{name}</Link></h4> */}
    <h4>{name}</h4>
    {species && <h4>Species: {species}</h4>}
    {status && <h4>Status: {status}</h4>}
    {image && <img src={image} />}
  </section>
);

List.propTypes = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  species: PropTypes.string,
  status: PropTypes.string,
  image: PropTypes.string,
};

export default List;
