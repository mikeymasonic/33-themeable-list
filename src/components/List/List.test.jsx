import React from 'react';
import { shallow } from 'enzyme';
import List from './List';

describe('List component', () => {
  it('renders List', () => {
    const wrapper = shallow(<List
      id="1"
      name="Rick Sanchez"
      species="Human"
      status="Alive"
      image="https://rickandmortyapi.com/api/character/avatar/1.jpeg"    
      />);
    expect(wrapper).toMatchSnapshot();
  });
});
