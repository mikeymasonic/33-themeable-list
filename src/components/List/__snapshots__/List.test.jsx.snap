// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders List 1`] = `
<section>
  <p>
    <Link
      to="detail/1"
    >
      Rick Sanchez
    </Link>
  </p>
  <p>
    Species: 
    Human
  </p>
  <p>
    Status: 
    Alive
  </p>
  <img
    src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
  />
</section>
`;
